integration_tests/test_rough_align.py

```diff
@@ -0,0 +1,382 @@
+
+    
+def test_montage_scape_stack(render, montage_scape_stack):
+    zvalues = render.run(renderapi.stack.get_z_values_for_stack, montage_scape_stack)
+    zs = [1020, 1021, 1022]
+    assert(set(zvalues)==set(zs)) 
+    
+def test_point_match_collection(render, rough_point_match_collection):
+    groupIds = render.run(renderapi.pointmatch.get_match_groupIds, rough_point_match_collection)
+    assert(('1020.0' in groupIds) and ('1021.0' in groupIds) and ('1022.0' in groupIds))
+
+def test_apply_rough_alignment_transform(render, montage_stack, test_do_rough_alignment, tmpdir_factory, prealigned_stack=None, output_stack=None):
+    ex1['render'] = render_params
+    ex1['montage_stack'] = montage_stack
+    ex1['lowres_stack'] = test_do_rough_alignment
+    ex1['prealigned_stack'] = None
+    ex1['output_stack'] = '{}_Rough'.format(montage_stack)
+    ex1['tilespec_directory'] = str(tmpdir_factory.mktemp('scratch'))
+    ex1['minZ'] = 1020
+    ex1['maxZ'] = 1022
+    ex1['scale'] = 0.1
+    
+    mod = ApplyRoughAlignmentTransform(input_data=ex1, args=[])
+    mod.run()
+
+    zend = ex1['maxZ']
+    zstart = ex1['minZ']
+
+    zvalues = render.run(renderapi.stack.get_z_values_for_stack, ex1['output_stack'])
+    zs = range(zstart, zend+1)
+
+    assert(set(zvalues) == set(zs))
+
+    apply_rough_alignment(render, 
+                          ex1['montage_stack'], 
+                          ex1['montage_stack'],
+                          ex1['lowres_stack'],
+                          ex1['tilespec_directory'],
+                          ex1['scale'],
+                          (1020,1020),
+                          consolidateTransforms=True)
+
+    # running again for code coverage
+    mod.run()
+
+    ex1['minZ'] = 1022
+    ex1['maxZ'] = 1020
+
+    mod = ApplyRoughAlignmentTransform(input_data=ex1, args=[])
+    with pytest.raises(RenderModuleException):
+        mod.run()
+
+
+# additional tests for code coverage
+
+def test_render_downsample_with_mipmaps(render, one_tile_montage, tmpdir_factory):
+    image_directory = str(tmpdir_factory.mktemp('rough'))
+    ex = {
+        "render": render_params,
+        "input_stack": one_tile_montage,
+        "image_directory": image_directory,
+        "imgformat": "png",
+        "scale": 0.1,
+        "minZ": -1,
+        "maxZ": -1,
+    }
+    ex['output_json'] = os.path.join(image_directory, 'output.json')
+
+    #stack_has_mipmaps = check_stack_for_mipmaps(render, ex['input_stack'], [1020])
+    tempjson = create_tilespecs_without_mipmaps(render, ex['input_stack'], 0, 1020)
+
+    mod = RenderSectionAtScale(input_data=ex, args=[])
+    mod.run()
+
+    with open(ex['output_json'], 'r') as f:
+        js = json.load(f)
+    
+    out_dir = os.path.join(image_directory, render_params['project'], js['temp_stack'], 'sections_at_0.1/001/0')
+    assert(os.path.exists(out_dir))
+    
+    files = glob.glob(os.path.join(out_dir, '*.png'))
+    for fil in files:
```

Comment:
shouldn't this create one specific file?

Reply:
No, each section has its own downsample image if that's what you are referring to. Also, individual tilespec file.